# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/TypedArrayObject.cpp
# Commit: a24fb00bab8e
# Full Hash: a24fb00bab8ea918e6751ddde6d7b71c12563d29
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1731039 - ArrayBufferOrView subclasses should null-check in unwrap() r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129092
# ==============================================================================

diff -r d6f67ce93bf2 -r a24fb00bab8e js/src/vm/TypedArrayObject.cpp
--- a/js/src/vm/TypedArrayObject.cpp	Thu Oct 21 17:45:50 2021 +0000
+++ b/js/src/vm/TypedArrayObject.cpp	Thu Oct 21 17:55:30 2021 +0000
@@ -2773,6 +2773,9 @@
 #undef INSTANTIATE
 
 JS::ArrayBufferOrView JS::ArrayBufferOrView::unwrap(JSObject* maybeWrapped) {
+  if (!maybeWrapped) {
+    return JS::ArrayBufferOrView(nullptr);
+  }
   auto* ab = maybeWrapped->maybeUnwrapIf<ArrayBufferObjectMaybeShared>();
   if (ab) {
     return ArrayBufferOrView::fromObject(ab);
